pkg/p2p/kadcast/maintainer.go

```diff
@@ -248,8 +249,10 @@ func (m *Maintainer) send(raddr net.UDPAddr, payload []byte) {
 
 	_ = conn.SetDeadline(time.Now().Add(10 * time.Second))
 
-	log.WithField("src", conn.LocalAddr().String()).
-		WithField("dest", raddr.String()).Traceln("Sending udp")
+	if log.Logger.GetLevel() == logrus.TraceLevel {
```

Comment:
This isn't necessary, is it? The call will just be ignored if we don't set the logger to capture `trace`.

Reply:
It is needed due to the `WithField` calls prior to Traceln. It was spotted by the cpu profiler.